SwiftWisdom/Core/UIKit/UIView+Extensions.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  UIView+Extensions.swift
+//  SwiftWisdom
+//
+//  Created by Alexander Persian on 12/29/16.
+//  Copyright © 2016 Intrepid. All rights reserved.
+//
+
+import UIKit
+
+public extension UIView {
+
+    /**
+     Zooms a view by scaling it from its current transform using the provided factor.
+
+     - parameter factor: Scaling factor. 1.0 = no change
+     */
+    func ip_zoom(by factor: CGFloat) {
+        transform = transform.scaledBy(x: factor, y: factor)
+    }
+
+    /**
+     Fades a view to alpha 0 over a specified duration. When the animation completes the alpha is returned to 1 to account for the possiblity of the view's `hidden` property being set to false without the use of a fade method.
+     
+     - parameter duration: Length of time for the fade. Default = 0.2s
+     */
+    func ip_fadeOut(over duration: TimeInterval = 0.2) {
+        UIView.animate(
```

Comment:
Since `fadeOut` is less dependent on the current state it doesn't need this check. If the view isn't visible this method just won't do anything. For `fadeIn` though we need to ensure that the view isn't hidden so that we can bring the alpha up progressively. If it's hidden the visual will be a pop to visible from the completion handler. Hope that makes sense.